racket/src/ChezScheme/c/segment.c

```diff
@@ -88,14 +88,14 @@ void *S_getmem(iptr bytes, IBOOL zerofill, UNUSED IBOOL for_code) {
 
   if ((addr = malloc(bytes)) == (void *)0) out_of_memory();
 
-  debug(printf("getmem(%p) -> %p\n", bytes, addr))
+  debug(printf("getmem(%ld) -> %p\n", bytes, addr))
```

Comment:
A more portable repair here is probably to use `TO_VOIDP` on the `addr` argument and keep `%p`.